core/src/main/scala/cats/data/Stream.scala

```diff
@@ -0,0 +1,713 @@
+package cats
+package data
+
+import cats.syntax.order._
+import scala.reflect.ClassTag
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+sealed abstract class Stream[A] { lhs =>
```

Comment:
Yeah, I'm definitely open to it if we can think of something good. I'd like to ensure that if we choose `Xyz` we also use `XyzT` for the transformer.